src/models/resource.ts

```diff
@@ -9,9 +9,15 @@ export type EnvironmentId = string & { readonly brand: unique symbol };
 
 export type OrganizationId = string & { readonly brand: unique symbol };
 
+/** Subtype of ConnectionType excluding ConnectionType.Platform, which we don't ever use here in extension-land. */
+export type UsedConnectionType =
+  | ConnectionType.Local
+  | ConnectionType.Ccloud
+  | ConnectionType.Direct;
```

Comment:
Could use `Exclude<ConnectionType, ConnectionType.Platform>` here instead of repeating each enum value